components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -212,7 +261,21 @@ func (sb *sensorBase) Stop(ctx context.Context, extra map[string]interface{}) er
 }
 
 func (sb *sensorBase) IsMoving(ctx context.Context) (bool, error) {
-	return sb.controlledBase.IsMoving(ctx)
+	if sb.velocities == nil {
+		return sb.controlledBase.IsMoving(ctx)
+	}
+	lin, err := sb.velocities.LinearVelocity(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	ang, err := sb.velocities.AngularVelocity(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	if lin.Y == 0.0 && ang.Z == 0.0 {
+		return false, nil
+	}
```

Comment:
There is a case in this driver where an orientation sensor can be provided but not a velocities sensor. We are ignoring that. Also this would mean we're not returning IsMoving correctly from the underlying wheeled base. I'll look into it.

Reply:
`IsMoving` for wheeled_base just returns `IsPowered` for all the motors and ignores the power percent. That doesn't work when using the control loop since it holds position, but maybe a better change is to fix `IsMoving` in wheeled base. I originally edited this because I was using `IsMoving` to determine when the tuning is done but I have a different solution now so I can worry about this in a different ticket if you'd prefer. but I also don't mind just changing it now. 

as for using only velocities == nil, that is the condition we use to check if we should start the loop or not on line 174 of this file. and since we are checking the linear and angular velocity we know that `velocities` has to support that. in the case that orientation is being used it will just call `IsMoving` of wheeled base. I _think_ this change is only relevant for using the control loop because is will never not be powered until you manually stop it, so you need to instead know if the velocities are 0 to know if it's moving or not

Reply:
but I haven't looked too closely at the situation in which you only have an orientation reporting sensor so I can look into that closer to understand how that would work with this change

Reply:
in the constructor for this driver, we also check for orientation reporting sensors and use only that in `spin.go`, so this change ignores that version of the sensor-controlled base completely. Since wheeled odometry is orientation reporting we can test what that's doing to he spin loop as well.